test/unit/cleanup-tests.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+# Copyright 2018 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+source $(dirname $0)/helper.sh
+
+readonly _CLEANUP_SCRIPT="tools/cleanup/cleanup.sh"
+source "$(dirname ${BASH_SOURCE})/../../${_CLEANUP_SCRIPT}"
+
+readonly _SUCCESS=0
+readonly _FAILURE=1
+
+readonly _FAKE_NIGHTLY_PROJECT_NAME="gcr.io/knative-nightly"
+readonly _FAKE_BOSKOS_PROJECT_NAME="gcr.io/fake-boskos-project"
+readonly _PROJECT_RESOURCE_YAML="ci/prow/boskos/resources.yaml"
+readonly _RE_PROJECT_NAME="knative-boskos-[a-zA-Z0-9]+"
+
+set -e
+cd ${REPO_ROOT_DIR}
+
+echo ">> Testing directly invoking cleanup script"
+
+test_function ${_FAILURE} "error: unknown option" ${_CLEANUP_SCRIPT} "--option-not-exist"
```

Comment:
I would prefer having a local helper function called `cleanup_script` instead of directly calling the script here. It works because we're `eval`ing the parameters passed, but it's not a straightforward readability.